pandas/core/indexers.py

```diff
@@ -133,34 +133,48 @@ def check_setitem_lengths(indexer, value, values) -> None:
 
     Returns
     -------
-    None
+    bool
+        Whether this is an empty listlike setting which is a no-op.
 
     Raises
     ------
     ValueError
         When the indexer is an ndarray or list and the lengths don't match.
     """
-    # boolean with truth values == len of the value is ok too
+    no_op = False
+
     if isinstance(indexer, (np.ndarray, list)):
-        if is_list_like(value) and len(indexer) != len(value):
-            if not (
-                isinstance(indexer, np.ndarray)
-                and indexer.dtype == np.bool_
-                and len(indexer[indexer]) == len(value)
-            ):
-                raise ValueError(
-                    "cannot set using a list-like indexer "
-                    "with a different length than the value"
-                )
+        # We can ignore other listlikes becasue they are either
+        #  a) not necessarily 1-D indexers, e.g. tuple
+        #  b) boolean indexers e.g. BoolArray
+        if is_list_like(value):
+            if len(indexer) != len(value):
+                if not (
+                    isinstance(indexer, np.ndarray)
+                    and indexer.dtype == np.bool_
+                    and len(indexer[indexer]) == len(value)
+                ):
+                    # boolean with truth values == len of the value is ok too
```

Comment:
```suggestion
                # boolean with truth values == len of the value is ok too
                if not (
                    isinstance(indexer, np.ndarray)
                    and indexer.dtype == np.bool_
                    and len(indexer[indexer]) == len(value)
                ):
```

(the diff is not really clear, but just suggesting to put the comment before the if check instead of after)